# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/rj-hassnain/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [ ]:
I chose the Refresh / Content Opportunity Scoring lane because it focuses on identifying pages that should be reviewed or refreshed based on
their performance signals. This lane interests me because it supports practical decision-making by helping prioritize which pages deserve attention
first instead of reviewing every page manually.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

In [ ]:
Decision: Which pages should be prioritized for content refresh?
Action: A content manager reviews the highest-ranked pages and decides whether to update, expand, or monitor them.
Cost of a wrong recommendation: If a healthy page is selected, time is wasted. If a declining page is missed, it may continue losing visibility and traffic.
Why ML helps: Machine learning can combine multiple signals such as impressions, CTR, content age, and average position to rank pages more effectively than a simple manual rule.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [6]:
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Total pages:", len(df))
print("Average CTR:", round(df["ctr"].mean(), 3))
print("Average impressions:", round(df["impressions_90d"].mean(), 2))

Total pages: 30000
Average CTR: 0.511
Average impressions: 5200.37


**3.1 Answer Cell**

The starter dataset contains **30,000 pages**. The average CTR is **0.511**, and the average number of impressions over 90 days is **5,200.37**. These values show that page performance varies across the dataset, making it suitable for building a ranking system that helps prioritize pages for content refresh.


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

In [8]:
This project provides decision support based on observed patterns in the available data. The results may identify pages that appear more likely to
benefit from review, but they do not prove that updating a page will improve its performance. The analysis does not predict Google's algorithm or
establish cause and effect.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.